src/processor/operator/projection.cpp

```diff
@@ -36,7 +36,10 @@ bool Projection::getNextTuplesInternal(ExecutionContext* context) {
         resultSet->multiplicity *=
             resultSet->getNumTuplesWithoutMultiplicity(discardedDataChunksPos);
     }
-    metrics->numOutputTuple.increase(1);
+    // The if statement is added to avoid the cost of calculating numTuples when metric is disabled.
+    if (metrics->numOutputTuple.enabled) [[unlikely]] {
+        metrics->numOutputTuple.increase(resultSet->getNumTuples(expressionsOutputDataChunksPos));
```

Comment:
I think it's better to track number of factorized tuple rather than flattened tuple but up to u.

Reply:
I'm less sure about this. We have to iterate this later I guess.